### ISSUE TITLE
Graphing Models

### ISSUE THREAD

**Issue by henripal** [pymc-devs/pymc#876]
The model graphing in PyMC is really useful! But PyDot/Graphviz doesn't play well with python 3 and produces graphs that are a little lackluster, and the graphing capability is absent from PyMC3.

I'm very much intermediate level but if someone else thought it was useful and someone was willing to help me out a little at the beginning, I'd be happy to try and contribute this feature to the project!

Some nice looking [Neo4j based graphing options](http://nicolewhite.github.io/neo4j-jupyter/hello-world.html?utm_content=bufferc17d9&utm_medium=social&utm_source=twitter.com&utm_campaign=buffer)

**Comment by aloctavodia** [comment]
I think it would be nice to have the graph rendered using the Kruschke-style-diagrams, [here](http://sumsar.net/blog/2013/10/diy-kruschke-style-diagrams/) there are some templates that could be really useful.

**Comment by twiecki** [comment]
@henripal Thanks for your interest, that would certainly be a valuable addition. The first to look into is how you could traverse the model graph as pymc3 does not have an explicit representation of parents and children.

**Comment by fonnesbeck** [comment]
BTW, if you are having PyDot issues in Python 3, try installing PyDot 2. 

```
pip install pydot2
```

**Comment by henripal** [comment]
I like the Kruschke-style graphs, as well!

Let me take a look at the code and see if I can figure out a way of doing this. 

(thanks for the pydot2 rec, too, I tried it and got it to work, albeit after a significant amount of tweaking)

**Comment by henripal** [comment]
Hey, so I looked hard at the code and follow the broad lines of the architecture, but find it to be rough going. I tried getting graphs from the Theano layer, to no avail, most likely for the reason @twiecki mentioned. 

I'm still interested, but need to hone the OOP side of my Python skills (open to suggestions on how to best do this!) before being able to actually solve this. I'll be back when I get enough XP to get to the next level :)

Thanks everyone for your time!
